### PR TITLE
Fix module dependency discovery to support msvs preprocessor

### LINKED ISSUE #45: Fix module dependency discovery to support microsoft prepocessor
Microsoft (R) C/C++ Optimizing Compiler Version 18.00.40629 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

cl : Command line warning D9002 : ignoring unknown option '-'
cl : Command line error D8003 : missing source filename
make[1]: *** Deleting file '/cygdrive/c/tmp/openj9-openjdk-jdk9_latest/build/windows-x86_64-normal-server-release/make-support/module-deps.gmk'

Fix c prepocessor flags when using microsoft visual studio toolchain.

### PR DESCRIPTION
**Pull request by vsebe**
Fixes #45 

Signed-off-by: Violeta Sebe <vsebe@ca.ibm.com>

### CODE REVIEW

**Comment by vsebe** [comment]
jenkins test sanity zlinux

File: make/common/Modules.gmk [hunk 1]
```diff
@@ -310,14 +310,21 @@ MODULE_DEPS_MAKEFILE := $(MAKESUPPORT_OUTPUTDIR)/module-deps.gmk
 
 MODULE_INFOS := $(call FindAllModuleInfos, *)
 
+ifeq ($(TOOLCHAIN_TYPE),gcc)
+  CPP_FLAGS += -P - <
+else ifeq ($(TOOLCHAIN_TYPE), microsoft)
+  CPP_FLAGS += -EP -nologo
+endif
```
Comment (keithc-ca) [inline make/common/Modules.gmk]: Modules.gmk is included many places: I don't think we should be modifying 'CPP_FLAGS' please invent a new macro to capture this difference. Also it needs to work if TOOLCHAIN_TYPE is neither gcc nor microsoft.

Comment (vsebe) [reply make/common/Modules.gmk]: @keithc-ca , fixed (might need to be revisited for other toolchains, e.g xlc)

**Comment by vsebe** [comment]
jenkins test sanity zlinux

File: make/common/Modules.gmk (hunk 1 above)
Comment (vsebe) [reply make/common/Modules.gmk]: @keithc-ca , @jdekonin  re-run build for fix, ready for review

**Pull request merged**